### ISSUE TITLE
`rxjs-no-unsafe-takeuntil` doesn't work for all cases.

### ISSUE THREAD

**Issue by emilio-martinez** [cartant/rxjs-tslint-rules#49]
I specifically applied these rules to our project interested in the `rxjs-no-unsafe-takeuntil` rule, knowing there are multiple sports where it's used. However, upon applying, to my suprise it only surfaced a few use cases.

I thought there might be some special-case handling depending on the operator, but I took a peek at the rule implementation and didn't see anything that stood out to me as to why the rule wouldn't work in our case.  

To give an example that I see in our codebase:

```ts
class MyClass {

   ...

   private _registerQuery (query: string): Query {
    if (this._queries.has(query)) {
      return this._queries.get(query)!;
    }

    const mql: MediaQueryList = this.mediaMatcher.matchMedia(query);
    const queryObservable = fromEventPattern<MediaQueryList>(
      (listener: Function) => { mql.addListener((e: MediaQueryList) => listener(e)); },
      (listener: Function) => { mql.removeListener((e: MediaQueryList) => listener(e)); }
    ).pipe(
      takeUntil(this._destroy),
      startWith(mql),
      map((nextMql: MediaQueryList) => ({ matches: nextMql.matches }))
    );

    const output = { observable: queryObservable, mql: mql };
    this._queries.set(query, output);
    return output;
  }
}
```

Happy to take more time to _actually_ debug afterwards, but thought I'd document in the meantime. The use cases that we have don't seem out of the ordinary, which is why I was surprised.

**Comment by cartant** [comment]
It won't check functions that are passed to the operator by name.

**Comment by cartant** [comment]
I really shouldn't comment on anything until I've had my morning coffee. The rule is positional, so it should be a bug.

**Comment by cartant** [comment]
Thanks for opening the issue. I've created a fixture for a repro and it appears that the rule has a problem with the `pipe` being appended to the `fromEventPattern` call. If the two are separated, the rule works as expected:

```ts
    const temp = fromEventPattern<MediaQueryList>(
      (listener: Function) => { mql.addListener((e: MediaQueryList) => listener(e)); },
      (listener: Function) => { mql.removeListener((e: MediaQueryList) => listener(e)); }
    );
    const queryObservable = temp.pipe(
      takeUntil(this._destroy),
      startWith(mql),
      map((nextMql: MediaQueryList) => ({ matches: nextMql.matches }))
    );
```

Should be pretty easy to fix. I hope.

**Comment by cartant** [comment]
This should be fixed in 4.4.2. As always, it was a copy-paste-related bug.
